From 52165b34c2fbcfe93f4cfabb616c2feb3ea7344f Mon Sep 17 00:00:00 2001
From: Jason Kridner <jdk@ti.com>
Date: Thu, 13 Oct 2011 23:50:35 -0400
Subject: [PATCH 3/4] am335x_evm: single-byte address EEPROM for board identifier

The BeagleBone has an EEPROM that uses a single byte address instead of a
2-byte address used by the EVM versions.  This simply adds a quick attempt
at reading the board identifier header with a 1-byte address if the header
signature isn't correct using a 2-byte address.

This is only build tested.  Having a script to program the expected EEPROM
contents would help me reduce my time to test this.

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 board/ti/am335x/evm.c |   14 ++++++++++++--
 1 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/board/ti/am335x/evm.c b/board/ti/am335x/evm.c
index af34bfd..f484a3c 100755
--- a/board/ti/am335x/evm.c
+++ b/board/ti/am335x/evm.c
@@ -496,8 +496,18 @@ int board_init(void)
 	}
 
 	if (header.magic != 0xEE3355AA) {
-		printf("Incorrect magic number in EEPROM\n");
-		goto err_out;
+		/* read the eeprom using i2c again, but use only a 1 byte address */
+		if (i2c_read(I2C_BASE_BOARD_ADDR, 0, 1, (uchar *)&header,
+								sizeof(header))) {
+			printf("Could not read the EEPROM; something fundamentally"
+				" wrong on the I2C bus.\n");
+			goto err_out;
+		}
+
+		if (header.magic != 0xEE3355AA) {
+			printf("Incorrect magic number in EEPROM\n");
+			goto err_out;
+		}
 	}
 
 	detect_daughter_board();
-- 
1.6.6.1

